modules/data.land/inst/doanload_NEON_multi_site.R

```diff
@@ -0,0 +1,111 @@
+#' Title generalized for download NEON files of multiSites. and automatically detect the most recent available NEON data and if data is problematic, try next year.
+#'
+#' @param settings a multisite settings object, should work with a single site setting. 
+#' @param main_store_dir the preferred path to store the downloaded NEON files. the log file and VEG_INFO should also be stored in that path
+#'
+#' @return VEG_INFO: a list object containing veg_info, site IDs of each site.
+#' @export log.txt file describing the progress of download
+
+Download_multiSites_NEON <- function(settings, main_store_dir){
+  #get NEON sites table
+  neonsites <- neonstore::neon_sites(api = "https://data.neonscience.org/api/v0", .token = Sys.getenv("NEON_TOKEN"))
+  
+  #output
+  VEG_INFO <- list()
+  
+  #log file
+  log_txt <- c()
+  
+  #starting FOR LOOP
+  for (i in 1:length(settings)) {
+    #match lon and lat
+    tempLat <- as.numeric(settings[[i]]$run$site$lat)
+    tempLon <- as.numeric(settings[[i]]$run$site$lon)
+    
+    betyneondist <- swfscMisc::distance(lat1 = tempLat, lon1 = tempLon, lat2 = neonsites$siteLatitude, lon2 = neonsites$siteLongitude)
+    index <- which(betyneondist == min(betyneondist))
+    tempNEON <- neonsites[index,]
+    
+    #grab data info from NEON
+    dataProducts <- tempNEON$dataProducts[[1]]
+    
+    #create subfolder
+    store_dir <- paste0(main_store_dir,"/",settings[[i]]$run$site$id)
+    dir.create(store_dir)
+    
+    #filter on date
+    if(length(which(dataProducts$dataProductCode=="DP1.10098.001")) == 0){
+      print(paste0("no data available for site: ", tempNEON$siteCode, ". jump to the next site!"))
+      log_txt <- c(log_txt, paste0("no data available for site: ", tempNEON$siteCode, ". jump to the next site!"))
+      next
+    }
+    NEON_dates <- lubridate::date(paste0(dataProducts$availableMonths[[which(dataProducts$dataProductCode=="DP1.10098.001")]], "-01"))
+    closeYear <- unique(lubridate::year(NEON_dates)[which(abs(lubridate::year(NEON_dates)-lubridate::year(settings$state.data.assimilation$start.date)) == min(abs(lubridate::year(NEON_dates)-lubridate::year(settings$state.data.assimilation$start.date))))])
+    target_date <- lubridate::date(paste0(closeYear,"-07-15"))
+    daysDiff <- abs(lubridate::days(lubridate::date(NEON_dates)-lubridate::date(target_date))@day)
+    
+    #search within 1 month
+    start_date <- ymd(as.Date(NEON_dates[which(daysDiff == min(daysDiff))]))
+    end_date <- start_date %m+% months(1)
```

Comment:
Definitely should NOT be hardcoding the search window!!

Reply:
As we've discussed verbally, you may need to look +/- multiple YEARS to make sure that you're seeing the full set of available veg plots, and then for each plot you want to find the "best" match in time. It's also been discussed (in an Issue?) that "best" not always be +/- in time, but there are instances (e.g. forward filtering) where you need to only look BACKWARDS in time. I believe that the solution proposed in an earlier meeting was to allow the upper and lower bounds of the search window to be user supplied arguments (and it's OK if both default to 1 mo), that would give the most flexibility for how the search was done.

Reply:
It sounds like not a easy fix but rather a redesign of the whole system, so I'd rather have a meeting with you to talk about this!

Reply:
search window still hard coded